dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java

```diff
@@ -306,41 +312,27 @@ private Container geContainerById(final String containerIdOrPath, final User use
-                final List<Contentlet> contentlets = personalizedContentletSet.stream().map(personalizedContentlet -> {
-                    try {
 
-                        // indexing
-                        final String contentletIdentifier = personalizedContentlet.getContentletId();
-                        final String personalization      = personalizedContentlet.getPersonalization();
-
-                        final Optional<Contentlet> contentletOpt = contentletAPI.findContentletByIdentifierOrFallback
-                                (personalizedContentlet.getContentletId(), mode.showLive, languageId, user, mode.respectAnonPerms);
-
-                        final Contentlet contentlet =  contentletOpt.isPresent()
-                                ? contentletOpt.get() : contentletAPI.findContentletByIdentifierAnyLanguage(personalizedContentlet.getContentletId());
+                final String currentPersonaTag = this.getCurrentPersonaTag(request);
+                final List<Map<String, Object>> cListAsMaps = Lists.newArrayList();
+                final List<Map<String, Object>> cListAsMaps2 = Lists.newArrayList();
 
-                        CollectionsUtils.computeSubValueIfAbsent(
-                                contentIdListByPersonalizationMap, personalization, contentletIdentifier,
-                                CollectionsUtils::add, (String key, String identifier)-> CollectionsUtils.list(identifier));
+                for (final PersonalizedContentlet personalizedContentlet : personalizedContentletSet) {
+                    final Contentlet contentlet = getContentlet(contentIdListByPersonalizationMap, personalizedContentlet);
 
-                        return contentlet;
-                    } catch (final DotContentletStateException e) {
-                        // Expected behavior, DotContentletState Exception is used for flow control
-                        return null;
-                    } catch (Exception e) {
-                        throw new DotStateException(e);
+                    if (contentlet == null) {
+                        continue;
                     }
-                }).filter(Objects::nonNull).collect(Collectors.toList());
-
-
-                final List<Map<String, Object>> cListAsMaps = Lists.newArrayList();
-                for (final Contentlet contentlet : contentlets) {
 
                     try {
                         final Map<String,Object> contentPrintableMap = ContentletUtil.getContentPrintableMap(user, contentlet);
                         contentPrintableMap.put("contentType", contentlet.getContentType().variable());
                         // contentPrintableMap.put("personalization", personalization); // todo: not sure if this will be needed by FOX
                         cListAsMaps.add(contentPrintableMap);
+
+                        if (personalizedContentlet.getPersonalization().equals(currentPersonaTag)) {
```

Comment:
I think we should Null check on the `personalizedContentlet.getPersonalization()` ?